overlord/patch/patch.go

```diff
@@ -0,0 +1,106 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package patch
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Level is the current implemented patch level of the state format and content.
+var Level = 1
+
+// patches maps from patch level L to the function that moves from L-1 to L.
+var patches = make(map[int]func(s *state.State) error)
+
+// SetCurrent sets state to the current patch level.
+func SetCurrent(s *state.State) {
```

Comment:
seems this one is not united tests.  Also not sure about the doc and name, they do not make that clear that is to be called only on a empty state (as the check and panic assure)